docs/proposals/14-advanced-scheduling.md

```diff
@@ -0,0 +1,77 @@
+
+Marking a seed as "unschedulable" prevents any new shoots to be scheduled onto this seed, due to the `NoSchedule` effect of the taint.
+
+Later, detecting unhealthy seeds and marking them as "unschedulable" could also be done automatically. This could be achieved by extending the `gardenlet` seed controller to react on certain seed conditions by adding certain taints with `NoSchedule` effects. Such taints could include for example:
+
+* `seed.gardener.cloud/not-ready`: The seed is `NotReady`. 
+* `seed.gardener.cloud/over-capacity`: The seed's capacity for shoots has been exceeded, see [Ensuring Seeds Capacity for Shoots Is Not Exceeded](13-automated-seed-management.md#ensuring-seeds-capacity-for-shoots-is-not-exceeded).
+
+### Evicting Shoots from Drained Seeds
+
+The next step in migrating shoots away from an unhealthy seed is actually draining the seed, which involves evicting all shoots from this seed. This could be achieved in one of the following ways:
+
+* By manually marking all seed shoots as orphaned, either by adding an `orphan` annotation or simply be deleting the `seedName` field in the shoot specification.
+* By a special `gardenctl` command, for example `gardenctl drain <seed>`, that would automate the action described above. It should also add the `seed.gardener.cloud/unschedulable` taint to the specified seed.
+
+**Note:** The behavior described here is similar to the behavior of the `kubectl drain <node>` command. This command adds a taint with a `NoSchedule` effect to the node, but doesn't add any taints with a `NoExecute` effect. Instead, it simply forcibly evicts all pods from the node.
+
+### Scheduling Orphaned Shoots to Healthy Seeds 
+
+The next step in migrating shoots away from an unhealthy seed is rescheduling all orphaned shoots onto healthy seeds by `gardener-scheduler`. The scheduler should ensure, among other things, that:
+
+* A shoot with the `orphan` annotation or with an empty `seedName` field is rescheduled.
+* A shoot is never scheduled onto a seed tainted with a taint with a `NoSchedule` effect (unless the shoot tolerates the taint).
+* The new seed capacity for shoots is not exceeded by adding a new shoot, see [Ensuring Seeds Capacity for Shoots Is Not Exceeded](13-automated-seed-management.md#ensuring-seeds-capacity-for-shoots-is-not-exceeded). This could perhaps also be achieved by automatically tainting the seed with a `NoSchedule` taint once its capacity is exceeded by the `gardenlet`.
+
+To reschedule a shoot, the scheduler updates the `seedName` field in the shoot specification with the name of the new seed. 
```

Comment:
I think, all of the requirements should already be fulfilled, if the previous sections are implemented.
So this section can probably be shortened/condensed to rather describe "Rescheduling Shoots" (scheduler acts also on existing shoots for which the `seedName` field has been unset).

Reply:
Yes, I will rewrite it as suggested.

Reply:
Rewritten as suggested and shortened considerably.